projects/fmcjesdadc1/src/app/fmcjesdadc1.c

```diff
@@ -280,8 +366,106 @@ int main(void)
 
 	// set up clock generator
 	status = ad9517_setup(&ad9517_device, ad9517_param);
-	if(status < 0)
+	if(status < 0) {
 		printf("Error ad9517_setup()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 0 power mode on */
+	status = ad9517_power_mode(ad9517_device, 0, 0);
+	if(status < 0) {
+		printf("Error channel 0 ad9517_power_mode()\n");
+		return FAILURE;
+	}
+	/* Set the channel 0 frequency to 250Mhz */
+	status = ad9517_frequency(ad9517_device, 0, 250000000);
+	if(status < 0) {
+		printf("Error channel 0 ad9517_frequency()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 1 power mode on */
+	status = ad9517_power_mode(ad9517_device, 1, 0);
+	if(status < 0) {
+		printf("Error channel 1 ad9517_power_mode()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 1 frequency to 250Mhz */
+	status = ad9517_frequency(ad9517_device, 1, 250000000);
+	if(status < 0) {
+		printf("Error channel 1 ad9517_frequency()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 4 power mode on */
+	status = ad9517_power_mode(ad9517_device, 4, 0);
+	if(status < 0) {
+		printf("Error channel 4 ad9517_power_mode()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 4 frequency to 250Mhz */
+	status = ad9517_frequency(ad9517_device, 4, 250000000);
+	if(status < 0) {
+		printf("Error channel 4 ad9517_frequency()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 5 power mode on */
+	status = ad9517_power_mode(ad9517_device, 5, 0);
+	if(status < 0) {
+		printf("Error channel 5 ad9517_power_mode()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 5 frequency to 250Mhz */
+	status = ad9517_frequency(ad9517_device, 5, 250000000);
+	if(status < 0) {
+		printf("Error channel 5 ad9517_frequency()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 6 power mode on */
+	status = ad9517_power_mode(ad9517_device, 6, 0);
+	if(status < 0) {
+		printf("Error channel 6 ad9517_power_mode()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 6 frequency to 250Mhz */
+	status = ad9517_frequency(ad9517_device, 6, 250000000);
+	if(status < 0) {
+		printf("Error channel 6 ad9517_frequency()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 7 power mode on */
+	status = ad9517_power_mode(ad9517_device, 7, 0);
+	if(status < 0) {
+		printf("Error channel 7 ad9517_power_mode()\n");
+		return FAILURE;
+	}
+
+	/* Set the channel 7 frequency to 250Mhz */
+	status = ad9517_frequency(ad9517_device, 7, 250000000);
+	if(status < 0) {
+		printf("Error channel 7 ad9517_frequency()\n");
+		return FAILURE;
+	}
+
+	status = ad9517_update(ad9517_device);
+	if(status < 0) {
+		printf("Error channel ad9517_update()\n");
+		return FAILURE;
+	}
+
+	status = ad9517_read(ad9517_device, AD9517_REG_PLL_READBACK, &reg_value);
+	if (status < 0) {
+		printf("AD9517 PLL not locked!\n");
+		return FAILURE;
+	} else
```

Comment:
nitpick: need brackets at `else` because `if` has brackets.

Reply:
Sure. Will do.

Reply:
actually `else` could be removed and code under `else` placed at the same level as the `if`

Reply:
yes, you are right. I will remove the `else` statement.

Reply:
Also, I think that the register value is what validates the PLL lock here.
```suggestion
	} else if (reg_value & AD9517_DIGITAL_LOCK_DETECT) {
```
Only a successful read status does not guarantee the PLL lock happened.

Reply:
You are right, I did check for the specific PLL lock bit at some point, but forgot to add it to the commit. Will add.